packages/flutter_tools/test/general.shard/fake_process_manager_test.dart

```diff
@@ -172,4 +175,149 @@ void main() {
       expect(stdout, 'stdout'.codeUnits);
     });
   });
+
+  group(FakeProcessManager, () {
+    late FakeProcessManager manager;
+
+    setUp(() {
+      manager = FakeProcessManager.empty();
+    });
+
+    group('start', () {
+      testWithoutContext('can run a fake command', () async {
+        manager.addCommand(const FakeCommand(command: <String>['faketool']));
+
+        final Process process = await manager.start(<String>['faketool']);
+        expect(await process.exitCode, 0);
+        expect(await utf8.decodeStream(process.stdout), isEmpty);
+        expect(await utf8.decodeStream(process.stderr), isEmpty);
+      });
+
+      testWithoutContext('outputFollowsExit delays stderr, stdout until after process exit', () async {
+        manager.addCommand(const FakeCommand(
+          command: <String>['faketool'],
+          stderr: 'hello',
+          stdout: 'world',
+          outputFollowsExit: true,
+        ));
+
+        final List<int> stderrBytes = <int>[];
+        final List<int> stdoutBytes = <int>[];
+
+        // Start the process.
+        final Process process = await manager.start(<String>['faketool']);
+        final StreamSubscription<List<int>> stderrSubscription = process.stderr.listen((List<int> chunk) { stderrBytes.addAll(chunk); });
+        final StreamSubscription<List<int>> stdoutSubscription = process.stdout.listen((List<int> chunk) { stdoutBytes.addAll(chunk); });
+
+        // Immediately after exit, no output is emitted.
+        await process.exitCode;
+        expect(utf8.decode(stderrBytes), isEmpty);
```

Comment:
Is it guaranteed that this code will run before the stdout/stderr is emitted? It looks like in the `FakeProcess` constructor this is set in `exitCode.then()`.

Reply:
We're awaiting `process.exitCode` right above this on line 213, so the fake process should have exited by then.

Reply:
To answer my own question, it is, because the Future created by `exitCode.then()` is then passed to `Stream.fromFuture`

Reply:
Good point. My attempt at removing that Future a while back is precisely what made me want to add these tests. Added a comment where we emit the Future to help clarify the intended behaviour since it's not at all obvious.